packages/sdk/test/integration/client_test.ts

```diff
@@ -863,4 +863,24 @@ describe.sequential('Client', function () {
       assert.equal(d1.toSortedJSON(), d2.toSortedJSON());
     }, task.name);
   });
+
+  it('Throw error when broadcasting unserializeable payload', async ({
```

Comment:
How about adding a test for a success case?

Reply:
Thank you for the suggestion. I've added a test case for successful broadcasting. 
However, I also wanted to add the following test case, which tests broadcasting between two clients. Unfortunately, the test code below doesn't work as expected because both `c1` and `c2`, provided by `withTwoClientsAndDocuments`, share the same `rpcAddr` . I believe this prevents them from behaving like separate clients in the test environment. Is there any way to make the following test case work?

```javascript
  it('Successfully broadcast serializeable payload', async ({ task }) => {
    await withTwoClientsAndDocuments<{ t: Text }>(async (c1, d1, c2, d2) => {
      const spy = vi.fn();

      const topic = 'test';
      const unsub = d2.subscribeBroadcastEvent(topic, spy);

      const payload = { a: 1, b: '2' };
      await c1.broadcast(d1, topic, payload);

      expect(spy.mock.calls.length).toBe(1);
      expect(spy.mock.calls[0][0]).toBe(topic);
      expect(spy.mock.calls[0][1]).toEqual(payload);

      unsub();
    }, task.name);
  });

```

Reply:
@sejongk 
Thank you, it works now! I tried both of the approaches you suggested. However, the first approach using `.sync()` didn't seem to work due to the uncertainty in the timing of when the broadcast event would arrive. So, I decided to go with the second approach. Instead of creating a new `withTwoClientsAndDocuments` function with `SyncMode.Realtime`, I added a new optional `syncMode` parameter. Here is the new function signature.

```javascript
async function withTwoClientsAndDocuments<T>(
  callback: (
    c1: Client,
    d1: Document<T>,
    c2: Client,
    d2: Document<T>,
  ) => Promise<void>,
  title: string,
  syncMode: SyncMode = SyncMode.Manual,
): Promise<void>
```

Additionally, I added a few more test cases to handle subscribing and unsubscribing.

Reply:
It seems that broadcast events are propagated through watchDocument, so it appears that testing isn't possible when real-time sync is not enabled, as the watchDocument stream isn't connected. In the Go implementation, the watch interface was exposed to users, allowing for event verification.(ref. https://github.com/yorkie-team/yorkie/issues/584) 

For future documentation, it would be helpful to clarify that broadcast events can only be received when sync mode is set to real-time.

Reply:
Oh, now I see why it didn't work before. I opened an issue regarding documentation [issue](https://github.com/yorkie-team/yorkie-team.github.io/issues/164), and will make sure to include the following limitation: Broadcast events can only be received when the sync mode is set to real-time.